drivers/avr/serial.c

```diff
@@ -52,7 +52,7 @@
 #                define EICRx_BIT (~(_BV(ISC30) | _BV(ISC31)))
 #                define SERIAL_PIN_INTERRUPT INT3_vect
 #            endif
-#        elif SOFT_SERIAL_PIN == E6
+#        elif defined(__AVR_ATmega32U4__) && SOFT_SERIAL_PIN == E6
```

Comment:
```suggestion
#        elif defined(__AVR_ATmega16U4__) || defined(__AVR_ATmega32U4__) && SOFT_SERIAL_PIN == E6
```

Reply:
`A || B && C`  is equivalent to `A || (B && C)`
This should be `(A || B) && C`

Reply:
Oops. Got sidetracked by something else before I hit the button on this one, completely forgot to add those.